File: solutions/228/228.summary-ranges.go
/*
 * @lc app=leetcode id=228 lang=golang
 *
 * [228] Summary Ranges
 */

// @lc code=start
import "strconv"

func summaryRanges(nums []int) []string {
    result := []string{}
    if (len(nums) == 0) {
        return result
    }

    start := nums[0]
    for i := 0; i < len(nums); i++ {
        if i == len(nums)-1 || nums[i]+1 != nums[i+1] {
            if nums[i] == start {
                result = append(result, strconv.Itoa(nums[i]))
            } else {
                result = append(result, strconv.Itoa(start) + "->" + strconv.Itoa(nums[i]))
            }
            if i != len(nums)-1 {
                start = nums[i+1]
            }
        }
    }
    return result
}
// @lc code=end

File: solutions/228/228.summary-ranges.php
/*
 * @lc app=leetcode id=228 lang=php
 *
 * [228] Summary Ranges
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums
     * @return String[]
     */
    function summaryRanges($nums) {
        if (!$nums) return [];

        $answer = [];
        $start = $nums[0];
        $diff = 1;

        for ($i = 1; $i < count($nums); $i++) {
            $cur = $nums[$i];
            if ($cur - $start === $diff) $diff++;
            else {
                if ($diff === 1) $answer[] = (string)$start;
                else $answer[] = (string)$start . "->" . (string)$nums[$i-1];
                $start = $cur;
                $diff = 1;
            }
        }

        if ($diff === 1) $answer[] = (string)$start;
        else $answer[] = (string)$start . "->" . (string)$nums[count($nums)-1];

        return $answer;
    }
}
// @lc code=end

